boilerplate/openshift/golang-osd-operator/csv-generate/common-generate-operator-bundle.py

```diff
@@ -310,6 +329,9 @@ def trim_index(index, kind, item):
 ## Process any additional resources
 # These don't go in the CSV; they just get copied into the bundle as is.
 for kind, docs in by_kind.items():
+    if kind not in BUNDLE_PERMITTED_RESOURCES:
+        print(f'Skipping resource of kind {kind}')
+        continue
```

Comment:
Let's error here to force the user to have all and only supported resources in their deploy folder. Otherwise they might expect them to do something. Especially since we've made it easy to test the pipeline locally, this shouldn't be an undue burden.

(And if we're going to do that, we could put this loop up top for efficiency)